gradle-plugin/src/main/java/com/squareup/anvil/plugin/AnvilPlugin.kt

```diff
@@ -163,6 +163,15 @@ open class AnvilPlugin : KotlinCompilerPluginSupportPlugin {
         project.dependencies.add("implementation", "$GROUP:annotations:$VERSION")
       }
     }
+
+    // Create a configuration for collecting CodeGenerator dependencies
+    val anvilConfiguration = project.configurations.maybeCreate("anvil").apply {
+      description = "This configuration is used for dependencies with Anvil CodeGenerator " +
+        "implementations."
+    }
+    // Make the kotlin compiler classpath extend it to pick up our extra generators
+    project.configurations.getByName(PLUGIN_CLASSPATH_CONFIGURATION_NAME)
+      .extendsFrom(anvilConfiguration)
```

Comment:
I didn't catch this earlier. I think it would be better not delaying to add the configuration. I'd move the whole block up to place where the Anvil extension is created.

Reply:
I don't think we can because it looks up the kotlin plugin classpath configuration, which means it needs to wait until the kotlin plugin is applied. We could try to swing it with something like `configureEach` but I was intentionally just deferring it to the plugin application callback. Thoughts?

Reply:
The configuration has to be created earlier, but the compiler plugin classpath needs to be extended here. Note that this block is only called in an afterEvaluate for multiplatform projects and that's too late. I'll update the PR. I have the change ready because I was testing and confirming that.

Reply:
Ahh gotcha. Sounds good 👍